rust/android/src/lib.rs

```diff
@@ -50,7 +45,21 @@ pub unsafe extern "C" fn Java_org_ZingoLabs_Zingo_RustFFI_00024Companion_initnew
         .into_string()
         .unwrap();
 
-    let seed = rustlib::init_new(server_uri, data_dir, chain_hint);
+    let chain_hint = CString::from(CStr::from_ptr(
```

Comment:
I believe `CStr::from_ptr` is unsafe.   Can we add assertions at this point to constrain possible values from this call?

One possibility is to mark the specific lines of this code which are unsafe, instead of the whole fn?

Reply:
I believe it's better to make the code marked `unsafe` be as specific as possible.   I think we can-and-should iteratively improve this.

Reply:
great comment. i think we should open a new PR for this after merge though as it applies to a lot of existing code and is a seperate matter